### PR TITLE
Determine select2 version from ACF instead of guessing from ACF version

### LINKED ISSUE #79: Possibly incomplete fix for select2 issue
I just realised that the fix for #26 we introduced with #45 is incomplete as things can possibly still fail as people can opt to stick with select2 v3 even with ACF5.6+.

`add_filter('acf/settings/select2_version', function(){ return 3; });`

See https://www.advancedcustomfields.com/blog/acf-pro-5-6-0-ui-ux/ and https://www.advancedcustomfields.com/resources/acf-settings/ for more Details.

So instead of deciding based on the ACF version we'll probably need to determine what is actually enqueued on the page with `acf_get_setting('select2_version')`, expose this client side as part of the JS config and decide upon this data on how to scrape inside the taxonomy scraper.

### PR DESCRIPTION
**Pull request by kraftner**
fixes #79 

This could probably still be improved as the case of ACF5 with select2 v3 is now not actually covered by the tests. But for that I am waiting for #103 because it will add the infrastructure to add a CPT for which we could then switch select2 to v3.

### CODE REVIEW

File: js/yoast-acf-analysis.js [hunk 1]
```diff
@@ -688,7 +688,7 @@ Scraper.prototype.scrape = function(fields){
 
         if( field.$el.find('.acf-taxonomy-field[data-type="multi_select"]').length > 0 ){
 
-            var select2Target = (helper.acf_version >= 5.6)?'select':'input';
+            var select2Target = (acf.select2.version >= 4)?'select':'input';
```
Comment (andizer) [inline js/yoast-acf-analysis.js]: Maybe, it is nicer to have a global `minimalRequiredSelect2Version` variable that contains the number `4`, that is easier to read than only using a magical `4` value.

**Comment by andizer** [comment]
CR done. I've one single suggestion.

File: js/yoast-acf-analysis.js (hunk 1 above)
Comment (kraftner) [reply js/yoast-acf-analysis.js]: Hm...The things is that
1. This isn't a minimal required version, but rather a switch to work with any select2 version.
2. The number is compared to `acf.select2.version`, so isn't it pretty obvious that this is the select2 version?

**Comment by moorscode** [comment]
Acceptance done

**Pull request merged**